Patch VTK to enable python 3.8 in VTK 8.2

kwryankrattiger opened the pull request:
Add VTK 8.2.1a, a patched VTK for Python 3.8 support.

Re-enable VisIt in the Data and Vis SDK CI.

Roughly based on patches from, which do not cleanly apply to 8.2.0, and some additional changes required for compilation.
https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6269
https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6275

CC: @tgamblin

kwryankrattiger committed:
VTK: Add patch for python 3.8 support

kwryankrattiger committed:
CI: Re-enable VisIt in CI

kwryankrattiger commented:
@spackbot fix style

kwryankrattiger commented:
Followup for: https://github.com/spack/spack/pull/38619

CC: @cyrush

eugeneswalker commented:
Did you mean to make changes to `share/spack/gitlab/cloud_pipelines/configs/packages.yaml` here?

kwryankrattiger commented:
> Did you mean to make changes to `share/spack/gitlab/cloud_pipelines/configs/packages.yaml` here?

Yes. Without that change it was concretizing the deprecated package in generate and then failing to build the deprecated python later.

kwryankrattiger committed:
Configure spec matrix for stack with VisIt

kwryankrattiger committed:
Add pugixml dep for 8.2.0

Pugixml seems to require VTK 8.2.0 as well

brugger1 commented:
@kwryankrattiger I don't have a gitlab account to follow that link. Can you paste the relevant portion in this PR?

eugeneswalker commented:
The three `vtk@8.2.1a` builds are failing in ECP DAV CI stack:
* [vtk/eyzjquq 8.2.1a gcc@=11.1.0 linux-ubuntu20.04-x86_64_v3 Data and Vis SDK](https://gitlab.spack.io/spack/spack/-/jobs/7909659)
```
...
1 error found in build log:
     232      CMake/vtkMPI.cmake:3 (find_package)
     233      CMake/FindHDF5.cmake:9 (include)
     234      CMake/vtkModuleMacros.cmake:891 (find_package)
     235      ThirdParty/hdf5/CMakeLists.txt:5 (vtk_module_third_party)
     236    This warning is for project developers.  Use -Wno-dev to suppress i
            t.
     237    
  >> 238    CMake Error at CMake/vtkModuleMacros.cmake:891 (find_package):
     239      By not providing "Findpugixml.cmake" in CMAKE_MODULE_PATH this pr
            oject has
     240      asked CMake to find a package configuration file provided by "pug
            ixml", but
     241      CMake did not find one.
     242    
     243      Could not find a package configuration file provided by "pugixml"
             with any
     244      of the following names:
```

kwryankrattiger commented:
@eugeneswalker 
I fixed this in a branch. I am not sure how VTK 8.2 was ever building because I didn't add the pugixml dep in my patch. I added it as a dependency in my testing branch and getting that to work first.

kwryankrattiger committed:
Make VTK and ParaView consistent on proj dep

danlipsa commented on var/spack/repos/builtin/packages/paraview/package.py:
```diff
@@ -253,7 +253,7 @@ class Paraview(CMakePackage, CudaPackage, ROCmPackage):
 
     # ParaView depends on proj@8.1.0 due to changes in MR
```
You may want to clarify this comment

kwryankrattiger replied:
good call 👍🏽

danlipsa commented:
Looks good after fixing the mentioned comment.

kwryankrattiger committed:
OpenMPI 3: provides MP support by default

kwryankrattiger committed:
Add more details on proj dep

kwryankrattiger committed:
Add python 3.8 to test mock repo

kwryankrattiger committed:
Remove testing includes

kwryankrattiger committed:
Patches to get VisIt to go

kwryankrattiger committed:
CI: Disable VisIt with GUI in DAV
